.github/workflows/build-and-deploy.yml

```
@@ -61,13 +61,16 @@
   create-release:
-    needs: build # This job depends on the build job
+    needs: [ build, upload-release-asset ] # This job depends on the build job
     runs-on: ubuntu-latest # Use the latest Ubuntu runner
-    if: github.event_name == 'push' && github.ref == 'refs/heads/main' # Only run this job on push events to the main branch
+    #if: github.event_name == 'push' && github.ref == 'refs/heads/main' # Only run this job on push events to the main branch
     steps:
       - name: Check out repository
         uses: actions/checkout@v4 # Check out the repository code
 
+      - name: Build war file
+        run: mvn install # Build the project to generate the WAR file
```

yamllint

[warning] too few spaces before comment

(comments)